### DISCUSSION TITLE
Consuming plugin.versions file

### DISCUSSION THREAD

**Discussion by Tanushree2904** [diffplug/blowdryer#37] [Q&A]
Hi Team,

I have another query regarding plugin.versions.

1. Let's say I've 10 plugins declared in my plugin.version file and I'm using only 6 in my project which is consuming the plugin.versions.
So when I compile, my project is expecting the remaining 4 versions as well (which are not needed in this project and I do not wish to add them). 
Also , I do not wish to include any configuration to bypass the extra plugins as I've hundreds to repos which will be consuming the plugins.versions file.

I need the below behaviour from plugins.version file:
Define all the plugins from all the repos at one place and when I consume this file in my project, I can define the plugins I need in my settings.xml and the unwanted plugins should not throw any error and will be ignored.
Is it possible? If so, how?

2. When I use '-DsetPluginVersions' flag to set my plugin versions, it updates the plugin versions in my settings.gradle but the formatting is not proper. It is formatted like below:

```
plugins {
id 'com.diffplug.blowdryerSetup' version '1.7.0'
}
```

**Expectation:** (Keep the existing formatting)
```
plugins {
    id 'com.diffplug.blowdryerSetup' version '1.7.0'
}
```

Thanks in Advance !

**Comment by nedtwigg** [comment]
> Define all the plugins from all the repos in one place

You will have to split up your `plugin.versions`. For example, the way we handle this at DiffPlug is we have

- https://github.com/diffplug/blowdryer-diffplug/blob/main/src/main/resources/plugin.versions
- https://github.com/diffplug/blowdryer-diffplug/blob/main/src/main/resources/plugin-kotlin.versions

And they get combined like this

https://github.com/diffplug/atplug/blob/4108c68ac750ba33968407b46ead009b476d2d61/settings.gradle#L28-L35

```gradle
blowdryerSetup {
	github 'diffplug/blowdryer-diffplug', 'tag', '7.0.2'
	//devLocal '../blowdryer-diffplug'
	setPluginsBlockTo {
		it.file 'plugin.versions'
		it.file 'plugin-kotlin.versions'
	}
}
```

> but the formatting is not proper

You have to put the indentation *in* the `.versions` file. Look at the `plugin.version` files I linked above, and you'll see that each line starts with indentation.

The `setPluginsBlockTo` is super simple, and that is on purpose.

https://github.com/diffplug/blowdryer/blob/c6c8056cd614eb4463f037429e35a2a2e49606f0/src/main/java/com/diffplug/blowdryer/BlowdryerSetup.java#L422-L457
